ranges/typedranges.nim

```diff
@@ -50,6 +50,13 @@ proc toRange*[T](a: seq[T]): Range[T] {.inline.} = toImmutableRange(a)
 
 converter toImmutableRange*[T](a: MutRange[T]): Range[T] {.inline.} = Range[T](a)
 
+proc hash*(x: Range): Hash =
+  var h: Hash = 0
+  when rangesGCHoldEnabled:
+    h = h !& hash(x.gcHold)
+  h = h !& hash(x.mLen)
```

Comment:
This is not correct. You should hash only the active part of the range (i.e. from `start` up to `mLen`). 
I also suggest copying the `hash` function defined for `openarray` that depends on a `hash` function defined over the `T` type stored in the range:

https://github.com/nim-lang/Nim/blob/devel/lib/pure/hashes.nim#L239

You can also directly reuse this function by calling `toOpenArray` on the range.